### PR TITLE
additions to #234

### PR DESCRIPTION
**Pull request by d-morrison**
This is a patch to your PR #234;  there were some changes that were easier to do than explain. In particular, I wanted to test out `vdiffr::expect_doppelganger()`, and provide some examples of how to use it. Please review the comments below for details.

### CODE REVIEW

File: DESCRIPTION [hunk 1]
```diff
@@ -31,7 +31,8 @@ Imports:
     tidyr,
     utils,
     tidyselect,
-    cli
+    cli,
+    purrr
```
Comment (d-morrison) [inline DESCRIPTION:35]: added `purrr` to dependencies to use `purrr::keep()`  to simplify `autoplot.pop_data()` lines 143-144.

File: R/autoplot.pop_data.R [hunk 2]
```diff
@@ -33,16 +33,18 @@ autoplot.pop_data <- function(
     type = "density",
     strata = NULL,
     ...) {
-  if (!is.element(strata, names(object))) {
+
+  if (!is.null(strata) && !is.element(strata, names(object))) {
```
Comment (d-morrison) [inline R/autoplot.pop_data.R:37]: we shouldn't check for missing strata variables if `strata = NULL`.

File: R/autoplot.pop_data.R [hunk 3]
```diff
@@ -33,16 +33,18 @@ autoplot.pop_data <- function(
...
     cli::cli_abort(
       class = "unavailable_strata",
       message = c(
-        "x" = paste0("The option {.var {strata}}",
-                     "for argument {.arg strata} does not exist"),
-        "i" = "Provide a column that exist in {.envvar object}"
+        x = "The variable {.var {strata}} specified by argument {.arg strata}
+        does not exist in {.arg object}.",
+        i = "Please choose a column that exists in {.arg object}."
```
Comment (d-morrison) [inline R/autoplot.pop_data.R:43]: please review these edits for guidance on future PRs.

File: R/autoplot.pop_data.R [hunk 4]
```diff
@@ -51,36 +53,37 @@ autoplot.pop_data <- function(
     cli::cli_abort(
       class = "unavailable_type",
       message = c(
-        "x" = paste0("Can't create plotting facets",
-                     " with the specified `type` = {.arg {type}}."),
-        "i" = "The `type` argument accepts 'density' or 'age-scatter' options."
+        x = "{.fn autoplot.pop_data} does not currently have an option for
```
Comment (d-morrison) [inline R/autoplot.pop_data.R:56]: note the use of `.fn` markup option

File: R/autoplot.pop_data.R [hunk 5]
```diff
@@ -51,36 +53,37 @@ autoplot.pop_data <- function(
...
     )
   }
 }
 
 age_scatter <- function(
     object,
-    strata = NULL) {
+    strata = NULL,
+    age_var = object %>% get_age_var(),
+    value_var = object %>% get_value_var()) {
```
Comment (d-morrison) [inline R/autoplot.pop_data.R:68]: we can extract the metadata once here and use it repeatedly below

File: tests/testthat/_snaps/autoplot.pop_data/autoplot.pop_data.density.svg
Comment (d-morrison) [inline tests/testthat/_snaps/autoplot.pop_data/autoplot.pop_data.density.svg:1]: Please see https://testthat.r-lib.org/reference/expect_snapshot_file.html for details about how these files work.

File: tests/testthat/test-autoplot.pop_data.R [hunk 6]
```diff
@@ -1,14 +1,86 @@
...
+              standardize = TRUE
+            ) %>%
+              autoplot(strata = "Country", type = "density") %>%
+              ggplot2::ggsave(
+                filename = tempfile(),
+                device = "svg",
+                width = 8,
+                height = 8
+              ) %>%
+              expect_snapshot_file(name = "autoplot.pop_data.density.svg")
```
Comment (d-morrison) [inline tests/testthat/test-autoplot.pop_data.R]: Please start including `expect_snapshot_file()` in tests for functions that output graphics or large tables.

File: DESCRIPTION [hunk 7]
```diff
@@ -43,7 +44,8 @@ Suggests:
     bookdown,
     ggbeeswarm,
     DT,
-    spelling
+    spelling,
+    svglite
```
Comment (d-morrison) [inline DESCRIPTION]: needed for `expect_snapshot_file()`, apparently

File: tests/testthat/test-autoplot.pop_data.R [hunk 8]
```diff
@@ -1,14 +1,68 @@
...
+          {
+            xs_data <- load_pop_data(
+              file_path = "https://osf.io/download//n6cp3/",
+              age = "Age",
+              id = "index_id",
+              value = "result",
+              standardize = TRUE
+            ) %>%
+              autoplot(strata = "Country", type = "density") %>%
+              vdiffr::expect_doppelganger(title = "density")
```
Comment (d-morrison) [inline tests/testthat/test-autoplot.pop_data.R:40]: please familiarize yourself with this function and use it to test our other graphics functions (before changing them).

**Review by d-morrison — COMMENTED** [review]
Please review the inline comments, and let me know if you have questions about anything I've done.

File: DESCRIPTION (hunk 1 above)
Comment (chrisorwa) [reply DESCRIPTION:35]: This is in order.

File: R/autoplot.pop_data.R (hunk 3 above)
Comment (chrisorwa) [reply R/autoplot.pop_data.R:43]: This is well-noted.

File: R/autoplot.pop_data.R (hunk 4 above)
Comment (chrisorwa) [reply R/autoplot.pop_data.R:56]: Well-noted.

File: tests/testthat/test-autoplot.pop_data.R (hunk 8 above)
Comment (chrisorwa) [reply tests/testthat/test-autoplot.pop_data.R:40]: Well-noted.

**Review by chrisorwa — APPROVED** [review]
All good!

**Pull request merged**